src/Operation.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Aidantwoods\SecureHeaders;
+
+interface Operation
+{
+    /**
+     * Transform the given set of headers
+     *
+     * @param HeaderBag $headers
+     * @return void
+     */
+    public function modify(HeaderBag $headers);
```

Comment:
I usually prefer to be as explicit as possible with these things ;-)

That and I feel that the PHP docs are very misleading with regard to how objects are passed. (Only true mention of this behaviour that I could find is contained in the equivalent of some change [notes](http://www.php.net/manual/en/migration5.oop.php), while the page on [function arguments](https://php.net/manual/functions.arguments.php) flat out contradicts their explaination.